lib/deploy.js

```diff
@@ -26,21 +27,24 @@ module.exports = function(opts) {
 		'Content-Type': 'application/json',
 		'Authorization': token
 	};
-
+	
 	logger.info("creating slug tarball");
+	logger.verbose("expecting process type " + processType);
 	logger.verbose("trying to read start script from Procfile");
 	return readFile(project+"/Procfile", { encoding: 'UTF-8' })
-		.then(function(procfile) { return procfile.match(/web:(.*)/)[1]; })
+		.then(function(procfile) { return procfile.match(new RegExp(processType + ":(.*)"))[1]; })
```

Comment:
I'm not very familiar with regex in general (and in Javascript in particular), so there might be a better way to accomplish what I'm trying to do here.